### PR TITLE
Add Food Processor tool

### PR DESCRIPTION
**Pull request by AlecWhite**
First of all, I have to say that this is my first time using github and doing a PR, and I don't know exactly what are all the things needed, how the workflow works and a million other details, so any help and direction is welcome.

I found that making oatmeal takes a lot of time, as well as flour and other meals, so I thought this would be useful. 

Add a food processor item, as well as it's recipe, deconstructions and spawn item group to kitchens.

If the PR is accepted I'll add flour/meal recipes, as I don't know what amount and type of changes/additions are fit a single PR .

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
It needs to be used as a tool in recipes. As it currently stands, it's just a battery storage object.

**Comment by AlecWhite** [comment]
Do you mean that I should add the recipes in the same PR?

**Comment by Rivet-the-Zombie** [comment]
Yes, otherwise the item would have no function.

Also, your jsons are wonky. The item itself is missing its category information, might be other errors too.

**Comment by AlecWhite** [comment]
Added cornmeal, oatmeal, flour, bone/tainted bone meal recipes. They use 20 charges and takes a fourth of the time it would take with a manual tool.

I think I also may have fixed all the errors in all the tid bits plus the styling using http://dev.narc.ro/cataclysm/format.html

**Comment by Rivet-the-Zombie** [comment]
Please feed your jsons to the [DDA web linting tool](http://dev.narc.ro/tools/format/) to remove any remaining astyle errors.

**Comment by AlecWhite** [comment]
Do you mean the whole json?

**Comment by Rivet-the-Zombie** [comment]
You can just feed your additions into it.

**Comment by AlecWhite** [comment]
I think I already fed all the additions to it, and then copypasted it to the jsons.

Do I need to do anything else?

**Comment by Rivet-the-Zombie** [comment]
You're missing a space in _data/json/item_groups.json.

And the linter is also choking on _data/json/items/tools.json:

```
02:57:28 ERROR: Format error at line 8246
02:57:28 <     "type": "TOOL",
02:57:28 >     "type": "TOOL",
02:57:28 
02:57:28 HINT: Canonical output for this block was:
02:57:28     "id": "food_processor",
02:57:28     "type": "TOOL",
02:57:28     "category": "tools",
02:57:28     "name": "food processor",
02:57:28     "description": "This is a kitchen appliance capable of slicing, chopping, shredding, grinding, pureeing and mixing.",
02:57:28     "weight": 3000,
02:57:28     "volume": 8,
02:57:28     "price": 50,
02:57:28     "to_hit": -1,
02:57:28     "bashing": 10,
02:57:28     "material": [ "iron" ],
02:57:28     "symbol": "%",
02:57:28     "color": "white",
02:57:28     "ammo": "battery",
02:57:28     "initial_charges": 100,
02:57:28     "max_charges": 100
02:57:28 Makefile:943: recipe for target 'obj/lint.cache' failed
```

Looks like the fields are out of order, and there's at least one missing space. The autolinter can fix those if you feed the blocks of code into it.

**Comment by AlecWhite** [comment]
Thank you for pointing out what the errors where. 
Fed all thee additions to the linting, am I missing something else?

**Comment by Rivet-the-Zombie** [comment]
`All checks have passed`

The linter likes it now!

**Comment by AlecWhite** [comment]
Awesome!

Thank you for your help Rivet.

File: data/json/recipes/other/tool.json [hunk 1]
```diff
@@ -128,5 +128,27 @@
...
+    "autolearn": true,
+    "qualities": [
+      { "id": "SCREW", "level": 1, "amount": 1 },
+      { "id": "HAMMER", "level": 3, "amount": 1 }
+    ],
+    "tools": [ [ [ "soldering_iron", 20 ], [ "toolset", 20 ] ] ],
+    "components": [
+      [ [ "scrap", 4 ] ],
+      [ [ "cable", 2 ] ],
+      [ [ "pot", 1 ] ]
```
Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: One can create food processor with just some metal scrap, a pair of cables and a pot? Without some sort of an electronic control unit to control all its functions? That seems very unrealistic to me.

Comment (AlecWhite) [inline data/json/recipes/other/tool.json]: The food processor I has three(2 speeds and one work-while-pressed button) buttons plus the on/off, I thought you could make those with just cables and primitive contact switches.

Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: But these contact switches must be deployed on some sort of a circuit board, not on a simple metal scrap, right? I'd say one or two pieces of `e_scrap` would suffice for both circuit board and contact switches.

Comment (AlecWhite) [inline data/json/recipes/other/tool.json]: Okay, added one e_scrap to the recipe.

Comment (Rivet-the-Zombie) [inline data/json/recipes/other/tool.json]: Maybe a small electric motor too?

Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: I'm in for SEM.

**Comment by Night-Pryanik** [comment]
Please add electronic scrap and small electric motor (if any) to deconstruction recipe too.

File: data/json/recipes/other/tool.json [hunk 2]
```diff
@@ -128,5 +128,29 @@
...
     ]
+  }, 
+  {
+    "result": "food_processor",
+    "type": "recipe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 4,
+    "skills_required": [ "fabrication", 4 ],
```
Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: I'm not sure about this, but if this recipe requires only 4 level of fabrication, then this line (`"skills_required": [ "fabrication", 4 ],`) is not needed?
By the way, we can add requirement of some level of electronics, as now we are using electric motor and electric scrap in the recipe.

Comment (AlecWhite) [inline data/json/recipes/other/tool.json]: I don't know if the difficulty is a modifier for crafting success, so you made left me thinking. Decided to check the recipe for the coffeemaker recipe and copied it's crafting, as I think it is more or less the same difficulty.

File: data/json/recipes/other/tool.json [hunk 3]
```diff
@@ -134,9 +134,8 @@
     "type": "recipe",
     "category": "CC_OTHER",
     "subcategory": "CSC_OTHER_TOOLS",
-    "skill_used": "fabrication",
-    "difficulty": 4,
-    "skills_required": [ "fabrication", 4 ],
+    "skill_used": "electronics",
+    "difficulty": 3,
```
Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: Please tell what do you want to implement.
Fabrication 4 _and_ electronics 3? Or just electronics 3?

Comment (AlecWhite) [inline data/json/recipes/other/tool.json]: Decided to look at the recipe of the coffeemaker as it is more or less a kitchen appliance, and modified the recipe to be in line with other already-in-game kitchen appliance. 
Althou I consider the foodprocessor to be more simpler than the coffeemaker, as it is just a blender but with different settings.

Comment (Night-Pryanik) [inline data/json/recipes/other/tool.json]: So let's compare: 

- coffeemaker: needs recipe books, needs only tool with `screw` quality, needs 10 charges of toolset or soldering iron, needs much more components, including `amplifier` and `element`, but no SEM.

- food processor: no need for recipe books, needs tools with `screw` and `hammer` qualities, needs 20 charges of toolset or soldering iron, needs much less components, but including SEM.

Comment (AlecWhite) [inline data/json/recipes/other/tool.json]: Comparing it with other kitchen appliance, now that you made me aware that they use the electronics skill, I like it. 
The food processor just chop up things raging from chunk-sized to finely grounded. I think is a rather simple machine, and I wanted to make it simple to craft too.

**Comment by AMurkin** [comment]
@AlecWhite Congrats with first merged PR

**Pull request merged**